pkg/server/api/bundle.go

```diff
@@ -0,0 +1,78 @@
+package api
+
+import (
+	"errors"
+
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/spire/proto/spire-next/types"
+	"github.com/spiffe/spire/proto/spire/common"
+)
+
+func BundleToProto(b *common.Bundle) (*types.Bundle, error) {
+	if b == nil {
+		return nil, errors.New("no bundle provided")
+	}
+
+	td, err := spiffeid.TrustDomainFromString(b.TrustDomainId)
+	if err != nil {
+		return nil, err
+	}
+	var x509Authorities []*types.X509Certificate
+	for _, rootCA := range b.RootCas {
+		x509Authorities = append(x509Authorities, &types.X509Certificate{
+			Asn1: rootCA.DerBytes,
+		})
+	}
+
+	var jwtAuthorities []*types.JWTKey
+	for _, key := range b.JwtSigningKeys {
+		jwtAuthorities = append(jwtAuthorities, &types.JWTKey{
+			PublicKey: key.PkixBytes,
+			KeyId:     key.Kid,
+			ExpiresAt: key.NotAfter,
+		})
+	}
+	return &types.Bundle{
+		TrustDomain:     td.String(),
+		RefreshHint:     b.RefreshHint,
+		SequenceNumber:  0,
+		X509Authorities: x509Authorities,
+		JwtAuthorities:  jwtAuthorities,
+	}, nil
+}
+
+func ProtoToBundle(b *types.Bundle) (*common.Bundle, error) {
+	if b == nil {
+		return nil, errors.New("no bundle provided")
+	}
+
+	td, err := spiffeid.TrustDomainFromString(b.TrustDomain)
+	if err != nil {
+		return nil, err
+	}
+
+	rootCas := []*common.Certificate{}
+	for _, certificate := range b.X509Authorities {
+		rootCas = append(rootCas, &common.Certificate{
+			DerBytes: certificate.Asn1,
```

Comment:
We should be validating that the incoming ASN.1 certificate data is well formed. See the helpers added in the AppendBundle PR.

Reply:
Right, I'm thinking about moving the helpers added in AppendBundle PR to this file. And then, modify AppendBundle to use `ProtoToBundle` instead of the individual helpers. 

Sounds good?

Reply:
Yes, although the `api` package is where we've been sticking the `*ToProto` and `ProtoTo*` methods.